### PR TITLE
#2694. Add syntax tests for different expressions. Part 3

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t01.dart [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+main() {
+  int? x = 2 > 1 ? 4 : null;
+
+  var list = [
+    ?x * 1,
+    ?x % 2,
+    ?x ~/ 3,
+    ?x * null,
+    ?x % null,
+    ?x % null,
```
Comment (eernstg) [inline LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t01.dart:57]: Perhaps?:
```suggestion
    ?x ~/ null,
```

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t01.dart [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  ];
+  Expect.listEquals([4, 0, 1], list);
+
+  var set = {
+    ?x * 1,
+    ?x % 2,
+    ?x ~/ 3,
+    ?x * null,
+    ?x % null,
+    ?x % null,
```
Comment (eernstg) [inline LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t01.dart:67]: Ditto

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t02.dart [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  C() : super(4);
+
+  void test() {
+    var list = [
+      ?super * 1,
+      ?super % 2,
+      ?super ~/ 3,
+      ?super * null,
+      ?super % null,
+      ?super % null,
```
Comment (eernstg) [inline LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t02.dart:61]: Perhaps?:
```suggestion
      ?super ~/ null,
```

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t02.dart [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    ];
+    Expect.listEquals([4, 0, 1], list);
+
+    var set = {
+      ?super * 1,
+      ?super % 2,
+      ?super ~/ 3,
+      ?super * null,
+      ?super % null,
+      ?super % null,
```
Comment (eernstg) [inline LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t02.dart:71]: Ditto

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t02.dart [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    };
+    Expect.setEquals({4, 0, 1}, set);
+
+    var map1 = {
+      ?super * 1: 1,
+      ?super % 2: 2,
+      ?super ~/ 3: 3,
+      ?super * null: 4,
+      ?super % null: 5,
+      ?super % null: 6,
```
Comment (eernstg) [inline LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t02.dart:81]: Ditto

File: LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t02.dart [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+      7: ?super * 1,
+      8: ?super % 2,
+      9: ?super ~/ 3,
+      10: ?super * null,
+      11: ?super % null,
+      12: ?super % null,
```
Comment (eernstg) [inline LanguageFeatures/Control-flow-collections/Null-aware-elements/syntax_multiplicative_A01_t02.dart:87]: Ditto

**Review by eernstg — COMMENTED** [review]
Looks good! Couple of typos.

**Comment by sgrekhov** [comment]
Thank you! Typos fixed. PTAL

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**